pkg/storage/provider_abort_action_test.go

```diff
@@ -0,0 +1,272 @@
+package storage_test
+
+import (
+	"testing"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/fixtures"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/fixtures/testusers"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/storage/internal/testabilities"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+	"github.com/go-softwarelab/common/pkg/to"
+	"github.com/stretchr/testify/require"
+)
+
+const NumberOfDesiredUTXOs = 32
+
+func TestAbortActionSuccess(t *testing.T) {
+	// given:
+	given, cleanup := testabilities.GivenCustomStorage(t, fixtures.StorageServerPrivKey, "dbstorage_test")
+	defer cleanup()
+
+	activeStorage := given.Provider().GORM()
+	faucet := given.Faucet(activeStorage, testusers.Alice)
+	faucet.TopUp(100_000)
+
+	createResult, err := activeStorage.CreateAction(
+		t.Context(),
+		testusers.Alice.AuthID(),
+		fixtures.DefaultValidCreateActionArgs(),
+	)
+	require.NoError(t, err)
+	testabilities.ThenDBState(t, activeStorage).AllOutputs(testusers.Alice).WithCount(1 + NumberOfDesiredUTXOs)
+	// when:
+	result, err := activeStorage.AbortAction(
+		t.Context(),
+		testusers.Alice.AuthID(),
+		wdk.AbortActionArgs{
+			Reference: to.Ptr(createResult.Reference),
+		},
+	)
+
+	// then:
+	require.NoError(t, err)
+	require.NotNil(t, result)
+	require.True(t, result.Aborted)
+
+	thenDBState := testabilities.ThenDBState(t, activeStorage)
+	thenDBState.HasUserTransactionByReference(testusers.Alice, createResult.Reference).
+		WithStatus(wdk.TxStatusFailed)
+	thenDBState.AllOutputs(testusers.Alice).WithCount(1)
+}
+
+func TestAbortActionTransactionNotOutgoing(t *testing.T) {
+	// given:
+	given, cleanup := testabilities.GivenCustomStorage(t, fixtures.StorageServerPrivKey, "dbstorage_test")
+
+	defer cleanup()
+
+	activeStorage := given.Provider().GORM()
+
+	given.Faucet(activeStorage, testusers.Alice).TopUp(100_000)
+	createdTx, _ := given.ActionCreatedAndSigned(activeStorage)
+	txID := createdTx.Inputs[0].SourceTxID
+	testabilities.ThenDBState(t, activeStorage).AllOutputs(testusers.Alice).WithCount(1 + NumberOfDesiredUTXOs)
+	// when:
+	result, err := activeStorage.AbortAction(
+		t.Context(),
+		testusers.Alice.AuthID(),
+		wdk.AbortActionArgs{
+			Reference: &txID,
+		},
+	)
+
+	// then:
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "failed to abort action: txDirectionInvalid: must be an outgoing transaction.")
```

Comment:
If you will create this public error (as I'm suggesting in other comment), then here you could use require.ErrorIs